File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s == t) return true;
        
        if(s.size() != t.size()) return false;
        
        vector<int> counter(26, 0);
        
        for(int i=0; i<s.size(); i++){
            counter[s[i]-'a']++;
        }
        
        for(int i=0; i<t.size(); i++){
            if(counter[t[i]-'a'] == 0) return false;
            else counter[t[i] - 'a']--;
        }
        
        
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False
        
        s1 = [0]*26
        t1 = [0]*26
        
        for i in range(len(s)):
            s1[ord(s[i])-ord('a')] += 1
            t1[ord(t[i])-ord('a')] += 1
            
        return s1 == t1